<launch>
  <arg name="uuv_name"/>
  <arg name="model_name" default="$(arg uuv_name)"/>
  <arg name="saturation" default="40"/>

  <arg name="Kp" default="10,10,10,10,10,10"/>
  <arg name="Kd" default="0,0,0,0,0,0"/>
  <arg name="Ki" default="1,1,1,1,1,1"/>

  <arg name="x" default="0"/>
  <arg name="y" default="0"/>
  <arg name="z" default="-20"/>
  <arg name="phi" default="0"/>
  <arg name="theta" default="0"/>
  <arg name="psi" default="0"/>

  <arg name="output_dir" default="$(find mur_control)/config"/>
  <arg name="config_file" default="$(find mur_control)/config/thruster_manager.yaml"/>
  <arg name="tam_file" default="$(find mur_control)/config/TAM.yaml"/>


  <group ns="$(arg uuv_name)">
  <!--
      <node pkg="uuv_control_utils"
            type="trajectory_marker_publisher.py"
            name="trajectory_marker_publisher"
            output="screen">
          <remap from="trajectory" to="dp_controller/trajectory"/>
          <remap from="waypoints" to="dp_controller/waypoints"/>
      </node>
-->
      <node pkg="mur_control"
          type="mur_altitude_controller.py"
          name="mur_altitude_controller"
          output="screen"><!--
          <remap from="odom" to="pose_gt"/>
          <remap from="trajectory" to="dp_controller/trajectory"/>
          <remap from="input_trajectory" to="dp_controller/input_trajectory"/>
          <remap from="waypoints" to="dp_controller/waypoints"/>
          <remap from="error" to="dp_controller/error"/>
          <remap from="reference" to="dp_controller/reference"/>
          <remap from="thruster_output" to="thruster_manager/input_stamped"/>-->
          <rosparam subst_value="true">
              x: [$(arg x)]
              y: [$(arg y)]
              z: [$(arg z)]
              phi: [$(arg phi)]
              theta: [$(arg theta)]
              psi: [$(arg psi)]
              Kp: [$(arg Kp)]
              Kd: [$(arg Kd)]
              Ki: [$(arg Ki)]
              saturation: $(arg saturation)
          </rosparam>
      </node>
  </group>
</launch>
